pkg/semtech-loramac/contrib/semtech_loramac.c

```diff
@@ -580,6 +594,28 @@ uint8_t semtech_loramac_join(uint8_t type)
     return SEMTECH_LORAMAC_JOIN_SUCCEEDED;
 }
 
+void semtech_loramac_request_link_check(void)
+{
+    _semtech_loramac_link_check_received = false;
+    MlmeReq_t mlmeReq;
+    mlmeReq.Type = MLME_LINK_CHECK;
+    LoRaMacMlmeRequest(&mlmeReq);
```

Comment:
I would try to use here something similar to the [mlme_confirm](https://github.com/RIOT-OS/RIOT/blob/master/pkg/semtech-loramac/contrib/semtech_loramac.c#L242) function. That is, sending a message to the LoRaMAC thread to process this MLME request. Otherwise, this won't be thread safe.

Reply:
(Since in this case, this function is called from main thread)

Reply:
I know that problem, it's the same story with the get/set functions. But for the moment, it's fine I think, until I finalize the refactoring I'm working on (as described in the initial comment).

Reply:
maybe [this branch](https://github.com/Inria-Chile/RIOT/blob/gnrc_lorawan/drivers/include/net/netdev2/lorawan.h) is helpful for you. It's an abandoned branch though

Reply:
Thanks, it helps certainly! That's indeed in the spirit of what I have in mind.